core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1094,11 +1111,11 @@ public IndexShardCacheEntity(IndexShard indexShard, QueryPhase queryPhase, Searc
              * results.*/
             final int expectedSizeInBytes = 512;
```

Comment:
The assumptions that make 512 a good number might not be valid any more. The stats requests are something like 40 bytes without the bounds. With the bounds they can get pretty big (utf8 encoding of a string) but probably are still fairly small.